src/wallet/wallet2.cpp

```diff
@@ -4999,6 +4999,276 @@ bool wallet2::get_tx_key(const crypto::hash &txid, crypto::secret_key &tx_key) c
   return true;
 }
 
+void wallet2::check_tx_key(const crypto::hash &txid, const crypto::secret_key &tx_key, const cryptonote::account_public_address &address, uint64_t &received, bool &in_pool, uint64_t &confirmations)
+{
+  // check if the given tx secret key actually matches with the tx public key
+  crypto::public_key tx_pub_key;
+  THROW_WALLET_EXCEPTION_IF(!crypto::secret_key_to_public_key(tx_key, tx_pub_key), error::wallet_internal_error, "Failed to compute tx pubkey");
+
+  // fetch tx pubkey from the daemon
+  COMMAND_RPC_GET_TRANSACTIONS::request req;
+  COMMAND_RPC_GET_TRANSACTIONS::response res;
+  req.txs_hashes.push_back(epee::string_tools::pod_to_hex(txid));
+  THROW_WALLET_EXCEPTION_IF(!net_utils::invoke_http_json("/gettransactions", req, res, m_http_client) || (res.txs.size() != 1 && res.txs_as_hex.size() != 1),
+    error::wallet_internal_error, "Failed to get transaction from daemon");
```

Comment:
The RPC call should use the appropriate lock.

Reply:
ok